src/Database/Bloodhound/Internal/Query.hs

```diff
@@ -1620,7 +1620,7 @@ functionScoreFunctionsPair (FunctionScoreSingle fn)
 functionScoreFunctionsPair (FunctionScoreMultiple componentFns) =
   ("functions", toJSON componentFns)
 
-fieldTagged :: Monad m => (FieldName -> Object -> m a) -> Object -> m a
+fieldTagged :: (Monad m, MonadFail m)=> (FieldName -> Object -> m a) -> Object -> m a
```

Comment:
I tried building this with stack-8.6.yaml, and it failed. I think you'd need to import `MonadFail` for that compiler version.

Reply:
```haskell
bloodhound          >     Not in scope: type constructor or class ‘MonadFail’
bloodhound          >     Perhaps you meant ‘MonadFix’ (imported from Bloodhound.Import)
bloodhound          >      |
bloodhound          > 1623 | fieldTagged :: (Monad m, MonadFail m)=> (FieldName -> Object -> m a) -> Object -> m a
bloodhound          >      |```

Reply:
Thanks for finding this. I also went through older versions and adjusted imports accordingly. I also bumped LTS versions for them, only GHC 8.0 had a major LTS version bumped, it compiles, so hopefully, it shouldn't be a problem.